Branch-SDK/Branch-SDK/ContentDiscoverer.m

```diff
@@ -0,0 +1,287 @@
+                if (!pathProperties.isSkipContentDiscovery) {
+                    NSArray *filteredKeys = [pathProperties getFilteredElements];
+                    if (filteredKeys == nil || filteredKeys.count == 0) {
+                        [self discoverViewContents:rootView contentData:contentDataArray contentKeys:contentKeysArray clearText:isClearText ID:@""];
+                    }
+                    else {
+                        contentKeysArray = filteredKeys.mutableCopy;
+                        [self discoverFilteredViewContents:contentDataArray contentKeys:contentKeysArray clearText:isClearText];
+                    }
+                }
+            } else if (cdManifest.referredLink != nil) { // else discover content if this session is started by a link click
+                [self discoverViewContents:rootView contentData:nil contentKeys:contentKeysArray clearText:YES ID:@""];
+            }
+            if (contentKeysArray != nil && contentKeysArray.count > 0) {
+                NSMutableDictionary *contentEventObj = [[NSMutableDictionary alloc] init];
+                [contentEventObj setObject:[NSString stringWithFormat:@"%f",[[NSDate date] timeIntervalSince1970]] forKey: TIME_STAMP_KEY];
+                if (cdManifest.referredLink != nil) {
+                    [contentEventObj setObject:cdManifest.referredLink forKey: REFERRAL_LINK_KEY];
+                }
+                
+                [contentEventObj setObject:[NSString stringWithFormat:@"/%@",lastViewController.class] forKey: VIEW_KEY];
+                [contentEventObj setObject:!isClearText? @"true":@"false" forKey: HASH_MODE_KEY];
+                [contentEventObj setObject:contentKeysArray forKey:CONTENT_KEYS_KEY];
+                if (contentDataArray != nil && contentDataArray.count > 0) {
+                    [contentEventObj setObject:contentDataArray forKey:CONTENT_DATA_KEY];
+                }
+                
+                [[BNCPreferenceHelper preferenceHelper]saveBranchAnalyticsData:contentEventObj];
+            }
+        }
+    }
+}
+
+
+
+- (void)discoverViewContents:(UIView *)rootView contentData:(NSMutableArray *)contentDataArray contentKeys:(NSMutableArray *)contentKeysArray clearText:(BOOL)isClearText ID:(NSString *)viewId {
+    if ([rootView isKindOfClass:UITableView.class]) {
+        NSArray * cells = ((UITableView *)rootView).visibleCells;
+        int cellCnt = -1;
+        for (UIView* cell in cells) {
+            cellCnt++;
+            NSString *format;
+            if (viewId.length > 0 ) {
+                format = @"-%d";
+            } else {
+                format = @"%d";
+            }
+            NSString *cellViewId = [viewId stringByAppendingFormat:format ,cellCnt];
+            [self discoverViewContents:cell contentData:contentDataArray contentKeys:contentKeysArray clearText:isClearText ID:cellViewId];
+        }
+    } else {
+        if ([rootView respondsToSelector:@selector(text)]) {
```

Comment:
is this supposed to be a check to see if `getContentText` returns something? Why not just call it?